tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -0,0 +1,287 @@
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""
+Test suite for testing RGW Process Restart with DTM enabled.
+"""
+import logging
+import multiprocessing
+import os
+import secrets
+import threading
+from multiprocessing import Queue
+from time import perf_counter_ns
+
+import pytest
+
+from commons import configmanager
+from commons import constants as const
+from commons.helpers.health_helper import Health
+from commons.helpers.pods_helper import LogicalNode
+from commons.params import LATEST_LOG_FOLDER
+from commons.params import TEST_DATA_FOLDER
+from commons.utils import assert_utils
+from commons.utils import support_bundle_utils
+from commons.utils import system_utils
+from config import CMN_CFG
+from config import DTM_CFG
+from config import HA_CFG
+from config.s3 import S3_CFG
+from conftest import LOG_DIR
+from libs.dtm.dtm_recovery import DTMRecoveryTestLib
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.s3.s3_rest_cli_interface_lib import S3AccountOperations
+from libs.s3.s3_test_lib import S3TestLib
+from scripts.s3_bench import s3bench
+
+
+# pylint: disable=too-many-instance-attributes
+class TestRGWProcessRestart:
+    """Test Class for RGW Process Restart."""
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class."""
+
+        cls.log = logging.getLogger(__name__)
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.master_node_list = []
+        cls.worker_node_list = []
+        for node in CMN_CFG['nodes']:
+            node_obj = LogicalNode(hostname=node["hostname"],
+                                   username=node["username"],
+                                   password=node["password"])
+            if node["node_type"].lower() == "master":
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.test_completed = False
+        cls.health_obj = Health(cls.master_node_list[0].hostname,
+                                cls.master_node_list[0].username,
+                                cls.master_node_list[0].password)
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/test_dtm_config.yaml")
+        cls.rgw_process = 'rgw'
```

Comment:
can this go to constant?

Reply:
We have added process names in hctl in constants. To avoid confusion, I thought better to keep process names inside container here.
Please suggest.

Reply:
We have added process names in hctl in constants. To avoid confusion, I thought better to keep process names inside container here.
Please suggest.

Reply:
I'm ok if this is the only file which will use it, if for multiple files, we can add it to constants.